File: fix_seo.js
const fs = require('fs');
const path = require('path');

const dir = 'c:\\Users\\bumbo\\Документы\\GitHub\\AntigravityAgents\\MrCar final';

function walkSync(currentDirPath, callback) {
    fs.readdirSync(currentDirPath).forEach(function (name) {
        var filePath = path.join(currentDirPath, name);
        var stat = fs.statSync(filePath);
        if (stat.isFile() && (filePath.endsWith('.html') || filePath.endsWith('.js'))) {
            callback(filePath, stat);
        } else if (stat.isDirectory() && name !== 'node_modules' && name !== '.git') {
            walkSync(filePath, callback);
        }
    });
}

let modifiedFiles = 0;

walkSync(dir, function (filePath) {
    // skip this script itself
    if (filePath.endsWith('fix_seo.js')) return;

    let content = fs.readFileSync(filePath, 'utf8');
    let original = content;

    // 1. Webasto wrong slug -> correct slug
    content = content.replace(/webasto-diagnostika/g, 'webasto-diagnostika');

    // 2. Trailing slash on privaatsus
    content = content.replace(/href="\/privaatsus\/"/g, 'href="/privaatsus"');
    content = content.replace(/href="https:\/\/www\.mrcar\.ee\/privaatsus\/"/g, 'href="https://www.mrcar.ee/privaatsus"');
    content = content.replace(/href="\/ru\/privaatsus\/"/g, 'href="/ru/privaatsus"');
    content = content.replace(/href="https:\/\/www\.mrcar\.ee\/ru\/privaatsus\/"/g, 'href="https://www.mrcar.ee/ru/privaatsus"');

    // 3. HTTP to HTTPS for mrcar.ee
    content = content.replace(/http:\/\/www\.mrcar\.ee/g, 'https://www.mrcar.ee');
    content = content.replace(/http:\/\/mrcar\.ee/g, 'https://www.mrcar.ee');

    if (content !== original) {
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`Updated: ${filePath}`);
        modifiedFiles++;
    }
});

console.log(`DONE. Modified ${modifiedFiles} files.`);

File: fix_seo.py
import os
import re

def fix_seo_issues(directory):
    html_files = []
    js_files = []
    for root, dirs, files in os.walk(directory):
        if 'node_modules' in root or '.git' in root or '.firebase' in root or '.gemini' in root:
            continue
        for file in files:
            if file.endswith('.html'):
                html_files.append(os.path.join(root, file))
            elif file.endswith('.js'):
                js_files.append(os.path.join(root, file))

    all_files = html_files + js_files

    for filepath in all_files:
        try:
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
        except Exception as e:
            print(f"Error reading {filepath}: {e}")
            continue

        original_content = content

        # 1. Fix 404: to-diagnostika -> tehobsluzhivanie-diagnostika
        content = content.replace('to-diagnostika', 'tehobsluzhivanie-diagnostika')

        # 2. Fix 301: Remove .html from internal links
        # Matches href="/something.html" or href="something.html"
        # Be careful not to replace external links ending in .html or canonical tags if they need .html (though cleanUrls implies they don't)
        # We will only target specific known pages to be safe, or just use regex for local links.
        # Let's target href="/..."
        content = re.sub(r'href=["\'](/[^"\']+)\.html["\']', r'href="\1"', content)

        # 3. Fix 301: Remove trailing slashes from language roots (except root /)
        # href="/ru/" -> href="/ru"
        # href="/en/" -> href="/en"
        content = re.sub(r'href=["\']/ru/["\']', r'href="/ru"', content)
        content = re.sub(r'href=["\']/en/["\']', r'href="/en"', content)

        # Let's also check for canonical links just in case they have .html or trailing slashes
        content = re.sub(r'href=["\'](https://www.mrcar.ee/[^"\']+)\.html["\']', r'href="\1"', content)
        content = re.sub(r'href=["\'](https://www.mrcar.ee/(ru|en))/["\']', r'href="\1"', content)

        if content != original_content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed SEO issues in: {filepath}")

if __name__ == "__main__":
    fix_seo_issues('.')
